sdks/android/app/src/main/java/org/mono/android/AndroidRunner.java

```diff
@@ -72,8 +73,19 @@ void copyAssetDir (AssetManager am, String path, String outpath) {
 			for (int i = 0; i < res.length; ++i) {
 				String fromFile = path + "/" + res [i];
 				String toFile = outpath + "/" + res [i];
+
+				InputStream fromStream;
```

Comment:
I would rather check for the type of the file, instead of relying on an exception being thrown. Something like `File.isDirectory` [1] and `File.isFile` [1] would be most helpful here.

[1] https://docs.oracle.com/javase/7/docs/api/java/io/File.html

Reply:
@luhenry that was my first idea but the only thing we get from the AssetManager is a stream and `File.isDirectory` actually tries to verify if the path exists _on disk_ which isn't the case here:

```
public boolean isDirectory()

Returns:
true if and only if the file denoted by this abstract pathname exists and is a directory; false otherwise
```